packages/components/src/components/as-category-widget/as-category-widget.spec.ts

```diff
@@ -0,0 +1,185 @@
+
+    beforeEach(async () => {
+      testWindow = new TestWindow();
+      element = await testWindow.load({
+        components: [CategoryWidget],
+        html: '<as-category-widget></as-category-widget>'
+      });
+    });
+
+    it('should render properly', async () => {
+      element.heading = 'Category Widget Example';
+      element.description = 'Description for Category Widget';
+      element.categories = exampleCategories;
+      element.showClearButton = true;
+      await testWindow.flush();
+
+      expect(element).toMatchSnapshot();
+    });
+
+    it('should not render header when showHeader is false', async () => {
+      element.heading = 'Category Widget Example';
+      element.description = 'Description for Category Widget';
+      element.showHeader = false;
+      await testWindow.flush();
+
+      expect(element).toMatchSnapshot();
+    });
+
+    it('should render clear button', async () => {
+      element.showClearButton = true;
+      element.categories = exampleCategories;
+      await testWindow.flush();
+
+      expect(element).toMatchSnapshot();
+    });
+
+    it('should render bars according to total value', async () => {
+      element.categories = exampleCategories;
+      element.useTotalPercentage = true;
+      await testWindow.flush();
+
+      expect(element).toMatchSnapshot();
+    });
+  });
+
+  describe('.getSelectedCategories', () => {
+    it('should return selectedCategories', () => {
+      const selectedCategories = ['Category 1'];
+      categoryWidget.selectedCategories = selectedCategories;
+
+      expect(categoryWidget.getSelectedCategories()).toEqual(selectedCategories);
+    });
+  });
+
+  describe('.clearSelection', () => {
+    beforeEach(() => {
+      spyOn(categoryWidget, '_onCategoriesChanged');
+    });
+
+    it('should clear selected categories', () => {
+      const selectedCategories = ['Category 1'];
+      categoryWidget.selectedCategories = selectedCategories;
+
+      categoryWidget.clearSelection();
+
+      expect(categoryWidget.getSelectedCategories()).toEqual([]);
+      expect(categoryWidget._onCategoriesChanged).toHaveBeenCalled();
+    });
+  });
+
+  describe('._isSelected', () => {
+    it('should return true if category is selected', () => {
+      const category = 'Category 1';
+      categoryWidget.selectedCategories = [category];
+
+      expect(categoryWidget._isSelected(category)).toBe(true);
+    });
+
+    it('should return false if category is not selected', () => {
+      expect(categoryWidget._isSelected('Fake Category')).toBe(false);
+    });
+  });
+
+  describe('._toggleCategory', () => {
+    beforeEach(() => {
+      spyOn(categoryWidget, '_onCategoriesChanged');
+    });
+
+    it('should select category if it is not selected', () => {
+      const category = { name: 'Category 1', value: 10, color: '#000' };
+      categoryWidget._toggleCategory(category);
+      expect(categoryWidget._onCategoriesChanged).toHaveBeenCalled();
```

Comment:
similar issue here, I think testing the real event would result in a more strong test, this way if we change the implementation (ie, renaming _onCategoriesChanged to _onCategoriesChange) the test will fail